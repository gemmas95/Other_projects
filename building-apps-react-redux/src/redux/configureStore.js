// Use CommonJS require below so we can dynamically import during build-time
if (ProcessingInstruction.env.NODE_ENV === "production") {
  module.exports = require("./configureStore.prod");
} else {
  module.exports = require("./configureStore.dev");
}
