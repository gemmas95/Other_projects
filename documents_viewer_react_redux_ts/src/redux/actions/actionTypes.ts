import IDocument from "../../models/document.interface";

// Documents actions
export const LOAD_DOCUMENTS: string = "LOAD_DOCUMENTS";
export const CREATE_DOCUMENT_SUCCESS: string = "CREATE_DOCUMENT_SUCCESS";
export const DELETE_DOCUMENT_SUCCESS: string = "DELETE_DOCUMENT_SUCCESS";
export const FILTER_DOCUMENTS_BY_TYPE: string = "FILTER_DOCUMENTS_BY_TYPE";

// Api calls actions
export const LOADING: string = "LOADING";
export const NO_LOADING: string = "NO_LOADING";

export const API_CALL_ERROR: string = "API_CALL_ERROR";

// Interfaces actions
export interface LoadDocumentsAction {
  type: typeof LOAD_DOCUMENTS;
  documents: IDocument[];
}

export interface CreateDocumentSuccessAction {
  type: typeof CREATE_DOCUMENT_SUCCESS;
  document: IDocument;
}

export interface DeleteDocumentSuccessAction {
  type: typeof DELETE_DOCUMENT_SUCCESS;
  documentId: string;
}

export interface FilterDocumentsByType {
  type: typeof FILTER_DOCUMENTS_BY_TYPE;
  typeFilter: string;
}

export interface LoadingAction {
  type: typeof LOADING;
}

export interface NoLoadingAction {
  type: typeof NO_LOADING;
}

export interface ApiCallErrorAction {
  type: typeof API_CALL_ERROR;
}

export type DocumentsActionTypes =
  | LoadDocumentsAction
  | CreateDocumentSuccessAction
  | DeleteDocumentSuccessAction
  | FilterDocumentsByType;

export type ApiCallActionTypes =
  | LoadingAction
  | NoLoadingAction
  | ApiCallErrorAction;

export type AppActions = DocumentsActionTypes | ApiCallActionTypes;
