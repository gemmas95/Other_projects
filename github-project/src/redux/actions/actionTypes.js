// Contributors Actions
export const LOAD_CONTRIBUTORS_SUCCESS = "LOAD COURSES_SUCCESS";

// Common Actions
export const LOADING = "LOADING";
export const NO_LOADING = "NO_LOADING";
