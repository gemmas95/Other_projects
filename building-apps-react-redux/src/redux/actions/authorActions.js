import * as types from "./actionTypes";
import * as authorApi from "../../api/authorApi";

export function loadAuthorsSuccess(authors) {
  return { type: types.LOAD_AUTHORS_SUCCESS, authors };
}

// THUNK MIDDLEWARE, we don't need to pass dispatch as an argument, redux thunk injects dispacth
export function loadAuthors() {
  return function (dispatch) {
    return authorApi
      .getAuthors()
      .then((authors) => {
        dispatch(loadAuthorsSuccess(authors));
      })
      .catch((error) => {
        throw error;
      });
  };
}
