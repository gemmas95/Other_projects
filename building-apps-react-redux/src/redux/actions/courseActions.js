import * as types from "./actionTypes";
import * as courseApi from "../../api/courseApi";

// Action creator, the only must is to have a type
export function createCourse(course) {
  return { type: types.CREATE_COURSE, course };
}

export function loadCourseSuccess(courses) {
  return { type: types.LOAD_COURSES_SUCCESS, courses };
}

// THUNK MIDDLEWARE, we don't need to pass dispatch as an argument, redux thunk injects dispacth
export function loadCourses() {
  return function (dispatch) {
    return courseApi
      .getCourses()
      .then((courses) => {
        dispatch(loadCourseSuccess(courses));
      })
      .catch((error) => {
        throw error;
      });
  };
}
