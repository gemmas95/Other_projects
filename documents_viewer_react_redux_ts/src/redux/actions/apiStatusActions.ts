import * as types from "./actionTypes";
import { AppActions } from "./actionTypes";

export function beginApiCall(): AppActions {
  return { type: types.LOADING };
}

export function endApiCall(): AppActions {
  return { type: types.NO_LOADING };
}

export function apiCallError(): AppActions {
  return { type: types.API_CALL_ERROR };
}
