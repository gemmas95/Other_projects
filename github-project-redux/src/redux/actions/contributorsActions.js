import * as types from "./actionTypes";
import { getContributors } from "../../api/contributorsApi";
import { apiCallError } from "../actions/apiStatusAction";

export function loadContributorsSuccess(contributors) {
  return { type: types.LOAD_CONTRIBUTORS_SUCCESS, contributors };
}
function loadingCall(commons) {
  return { type: types.LOADING, commons };
}
function loadingSuccess(commons) {
  return { type: types.NO_LOADING, commons };
}

export function loadContributors(data) {
  // console.log("this is DISPACTH CONTRIBUTORSACTIONS", dispatch);
  return function (dispatch) {
    dispatch(loadingCall());
    // dispatch(beginApiCall());

    return getContributors(data)
      .then((response) => response.json())

      .then((contributors) => {
        dispatch(loadContributorsSuccess(contributors));
        dispatch(loadingSuccess());
      })
      .catch((error) => {
        dispatch(apiCallError(error));
        console.error(error);
        throw error;
      });
  };
}

export function sortByLessContributions(contributors) {
  console.log("ai que LLEGAAAA ....", contributors);

  return function (dispatch) {
    dispatch({
      type: types.FILTER_CONTRIBUTORS_BY_LESS_CONTRIBUTIONS,
      contributors,
      /*       contributors: [
        ...contributors.sort((a, b) => {
          console.log("this is a", a, "and this is b", b);
          return a.contributions - b.contributions;
        }),
      ], */
    });
    console.log("ai pollito....", contributors);
  };

  /*  return {
    ...state,
    contributors:
      state.contributors.length > 0
        ? [
            ...state.contributors.sort(
              (a, b) => a.contributions - b.contributions
            ),
          ]
        : state.contributors,
  }; */
}

export function sortByMoreContributions(contributors) {
  return function (dispatch) {
    dispatch({
      type: types.FILTER_CONTRIBUTORS_BY_MORE_CONTRIBUTIONS,
      contributors,
    });
  };
}

export function sortByNameAscending(contributors) {
  return function (dispatch) {
    dispatch({
      type: types.FILTER_CONTRIBUTORS_BY_ASCENDING_NAME,
      contributors,
    });
  };
}

export function sortByNameDescending(contributors) {
  return function (dispatch) {
    dispatch({
      type: types.FILTER_CONTRIBUTORS_BY_DESCENDING_NAME,
      contributors,
    });
  };
}
