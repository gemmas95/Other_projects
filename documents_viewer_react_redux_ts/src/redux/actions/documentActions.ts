import * as types from "./actionTypes";
import { AppActions } from "./actionTypes";
import IDocument from "../../models/document.interface";
import { Dispatch } from "react";

// Imports of fake api calls
import {
  getDocuments,
  saveDocument,
  deletingDocument,
} from "../../api/documentAPI";

import { beginApiCall, apiCallError, endApiCall } from "./apiStatusActions";

// Action types from action creators

export function loadDocumentSuccess(documents: IDocument[]): AppActions {
  return { type: types.LOAD_DOCUMENTS, documents };
}

export function createDocumentSuccess(document: IDocument): AppActions {
  // New documents it's saved in action property document
  return { type: types.CREATE_DOCUMENT_SUCCESS, document };
}

export function removeDocument(documentId: string): AppActions {
  return { type: types.DELETE_DOCUMENT_SUCCESS, documentId };
}

export function filterDocumentsByType(typeFilter: string): AppActions {
  return { type: types.FILTER_DOCUMENTS_BY_TYPE, typeFilter };
}

// Action creators

export function loadDocuments() {
  return async function (dispatch: Dispatch<AppActions>) {
    dispatch(beginApiCall());
    return await getDocuments()
      .then((documents) => {
        dispatch(loadDocumentSuccess(documents));
        dispatch(endApiCall());
      })
      .catch((error) => {
        dispatch(apiCallError());
        console.error(error);
        throw error;
      });
  };
}

// Notifications with toast to improve user experience could be handled here or in the component. I don't want
// to do action creators and the functions that handle them  too verbose so notifications are handled in components
export function createDocument(document: IDocument) {
  return async function (dispatch: Dispatch<AppActions>) {
    dispatch(beginApiCall());
    return await saveDocument(document)
      .then((savedDocument) => {
        dispatch(createDocumentSuccess(savedDocument));
        dispatch(endApiCall());
      })
      .catch((error) => {
        dispatch(apiCallError());
        console.error(error);
        throw error;
      });
  };
}

// Here I delete document optimistically for a better user experience
export function deleteDocument(id: string) {
  return async function (dispatch: Dispatch<AppActions>) {
    dispatch(removeDocument(id));
    return await deletingDocument(id).catch((error) => {
      console.error(error);
      throw error;
    });
  };
}

export function filterDocuments(type: string) {
  return function (dispatch: Dispatch<AppActions>) {
    dispatch(filterDocumentsByType(type));
  };
}
