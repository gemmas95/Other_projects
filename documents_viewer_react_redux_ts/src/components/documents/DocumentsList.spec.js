import React from "react";
import DocumentList from "./DocumentsList";
import { BrowserRouter } from "react-router-dom";
import renderer from "react-test-renderer";

import { Provider } from "react-redux";
import configureStore from "redux-mock-store";

// import { render } from "@testing-library/react";

// Action
import { deleteDocument } from "../../redux/actions/documentActions";
import { cleanup } from "@testing-library/react";

const mockStore = configureStore([]);

describe("DocumentList", () => {
  // const deleteDocument = () => {};
  const currentDocuments = [{}];
  let store;

  let DocumentListTree;
  beforeEach(() => {
    store = mockStore({
      myState: "sample text",
    });

    store.dispatch = jest.fn();

    DocumentListTree = renderer.create(
      <Provider store={store}>
        <BrowserRouter>
          <DocumentList
            deleteDocument={deleteDocument}
            currentDocuments={currentDocuments}
          />
        </BrowserRouter>{" "}
      </Provider>
    );
  });
  afterEach(() => {
    cleanup();
  });
  // It gives an error in console with unique key even though there is declared in child component 'DocumentItem'
  xit("should match snapshot", () => {
    expect(DocumentListTree.toJSON()).toMatchSnapshot();
  });

  /*   xit("should match snapshotsda", () => {
    const mockFn = jest.fn();
    const handleDelete = new mockFn();

    handleDelete();
    expect(deleteDocument).toHaveBeenCalled();
  }); */
});
