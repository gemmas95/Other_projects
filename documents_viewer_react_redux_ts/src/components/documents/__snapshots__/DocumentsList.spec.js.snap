// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DocumentList should match snapshot 1`] = `
Array [
  <ul
    className="document__container"
  >
    <li
      className="document__item"
    >
      <div>
        <a
          aria-current={null}
          href="/documents/undefined"
          onClick={[Function]}
        >
          <p />
        </a>
        <div>
          <p />
          <p />
        </div>
      </div>
      <button
        data-testid="button"
        onClick={[Function]}
      >
        Delete
      </button>
    </li>
  </ul>,
  " ",
  <div
    className="Toastify"
  />,
  " ",
]
`;
